lib/docs/scrapers/cmake.rb

```diff
@@ -13,51 +13,66 @@ class Cmake < UrlScraper
     options[:title] = false
     options[:root_title] = 'CMake Reference Documentation'
     options[:skip] = %w(release/index.html genindex.html search.html)
-    options[:skip_patterns] = [/\Agenerator/, /\Ainclude/, /\Arelease/]
+    options[:skip_patterns] = [/\Agenerator/, /\Acpack_gen/, /\Ainclude/, /\Arelease/]
 
     options[:attribution] = <<-HTML
-      &copy; 2000&ndash;2018 Kitware, Inc. and Contributors<br>
+      &copy; 2000&ndash;2019 Kitware, Inc. and Contributors<br>
       Licensed under the BSD 3-clause License.
     HTML
 
+    version '3.15' do
+      self.release = '3.15.2'
+      self.base_url = "https://cmake.org/cmake/help/v#{self.version}/"
+    end
+
+    version '3.14' do
+      self.release = '3.14.6'
+      self.base_url = "https://cmake.org/cmake/help/v#{self.version}/"
+    end
+
+    version '3.13' do
+      self.release = '3.13.5'
+      self.base_url = "https://cmake.org/cmake/help/v#{self.version}/"
+    end
+
     version '3.12' do
-      self.release = '3.12.1'
-      self.base_url = 'https://cmake.org/cmake/help/v3.12/'
+      self.release = '3.12.4'
+      self.base_url = "https://cmake.org/cmake/help/v#{self.version}/"
     end
 
     version '3.11' do
       self.release = '3.11.4'
-      self.base_url = 'https://cmake.org/cmake/help/v3.11/'
+      self.base_url = "https://cmake.org/cmake/help/v#{self.version}/"
     end
 
     version '3.10' do
       self.release = '3.10.3'
-      self.base_url = 'https://cmake.org/cmake/help/v3.10/'
+      self.base_url = "https://cmake.org/cmake/help/v#{self.version}/"
     end
 
     version '3.9' do
       self.release = '3.9.6'
-      self.base_url = 'https://cmake.org/cmake/help/v3.9/'
+      self.base_url = "https://cmake.org/cmake/help/v#{self.version}/"
     end
 
     version '3.8' do
       self.release = '3.8.2'
-      self.base_url = 'https://cmake.org/cmake/help/v3.8/'
+      self.base_url = "https://cmake.org/cmake/help/v#{self.version}/"
     end
 
     version '3.7' do
       self.release = '3.7.2'
-      self.base_url = 'https://cmake.org/cmake/help/v3.7/'
+      self.base_url = "https://cmake.org/cmake/help/v#{self.version}/"
     end
 
     version '3.6' do
       self.release = '3.6.3'
-      self.base_url = 'https://cmake.org/cmake/help/v3.6/'
+      self.base_url = "https://cmake.org/cmake/help/v#{self.version}/"
     end
 
     version '3.5' do
       self.release = '3.5.2'
-      self.base_url = 'https://cmake.org/cmake/help/v3.5/'
+      self.base_url = "https://cmake.org/cmake/help/v#{self.version}/"
```

Comment:
Could this be extracted to the root of the class?

Reply:
Nope, if you move it to the root of the class `self.version` will not evaluate to the correct version, causing all versions to get a `base_url` of `https://cmake.org/cmake/help/v/`.